web-common/tailwind.config.ts

```diff
@@ -0,0 +1,125 @@
+import type { Config } from "tailwindcss";
+import {
+  LightnessMap,
+  ThemeColorKind,
+  defaultPrimaryColors,
+  defaultSecondaryColors,
+  mutedColors,
+} from "./src/features/themes/color-config";
+
+/**
+ *
+ * @param colorMap
+ * @param colorName
+ * @returns
+ */
+function addThemeColorsAsVarRefs(
+  colorMap: LightnessMap,
+  themeColorKind: ThemeColorKind,
+) {
+  return Object.fromEntries(
+    Object.keys(colorMap).map((colorNum) => {
+      return [
+        `${themeColorKind}-${colorNum}`,
+        `var(--color-${themeColorKind}-${colorNum})`,
+      ];
+    }),
+  );
+}
+
+function initializeDefaultColorsVars(
+  colorMap: LightnessMap,
+  colorName: ThemeColorKind,
+) {
+  const colorVars: [string, string][] = Object.entries(colorMap).map(
+    ([colorNum, colorCssString]) => [
+      `--color-${colorName}-${colorNum}`,
+      colorCssString,
+    ],
+  );
+  return Object.fromEntries(colorVars);
+}
+
+export default {
+  // need to add this for storybook
+  // https://www.kantega.no/blogg/setting-up-storybook-7-with-vite-and-tailwind-css
+  content: ["./index.html", "./src/**/*.{js,ts,jsx,tsx,svelte}"],
+  /** Once we have applied dark styling to all UI elements, remove this line */
+  darkMode: "class",
+  theme: {
+    extend: {
+      colors: {
+        ...addThemeColorsAsVarRefs(defaultPrimaryColors, "primary"),
+        ...addThemeColorsAsVarRefs(defaultSecondaryColors, "secondary"),
+        ...addThemeColorsAsVarRefs(mutedColors, "muted"),
+
+        border: "hsl(var(--border) / <alpha-value>)",
+        input: "hsl(var(--input) / <alpha-value>)",
+        ring: "hsl(var(--ring) / <alpha-value>)",
+        background: "hsl(var(--background) / <alpha-value>)",
+        foreground: "hsl(var(--foreground) / <alpha-value>)",
+        primary: {
+          DEFAULT: "hsl(var(--primary) / <alpha-value>)",
+          foreground: "hsl(var(--primary-foreground) / <alpha-value>)",
+        },
+        secondary: {
+          DEFAULT: "hsl(var(--secondary) / <alpha-value>)",
+          foreground: "hsl(var(--secondary-foreground) / <alpha-value>)",
+        },
```

Comment:
These are pulling from CSS variables in `app.css`. Can you make sure these ShadCN variables reflect the theme as well?